src/state/application/hooks.ts

```diff
@@ -88,9 +88,22 @@ export function useToggleModal(modal: ApplicationModal): () => void {
   return useCallback(() => dispatch(setOpenModal(isOpen ? null : modal)), [dispatch, modal, isOpen])
 }
 
-export function useCloseModal(): () => void {
+export function useCloseModal() {
   const dispatch = useAppDispatch()
-  return useCallback(() => dispatch(setOpenModal(null)), [dispatch])
+  const currentlyOpenModal = useAppSelector((state: AppState) => state.application.openModal)
+  return useCallback(
+    (modal?: ApplicationModal) => {
+      // Check if modal is defined and either matches the currently open modal
+      // or is not a valid instance of ApplicationModal enum (i.e., if it is passed a click event)
+      if ((modal && currentlyOpenModal === modal) || (modal && !Object.values(ApplicationModal).includes(modal))) {
```

Comment:
in what cases would it pass a click event? also the function takes an application modal or undefined as the parameter so how would this happen?

Reply:
The `@reach/dialog` `DialogOverlay` component is mistyped internally and claims to not send a click event on `onDismiss`, but it actually does send one. This caused a bug (which e2e tests caught!!) and means we need to be extra defensive here (or could do it in the onDismiss handler too).

Reply:
moving the handling of this to the onDismiss handler